.github/workflows/publish-to-pypi.yml

```diff
@@ -5,22 +5,37 @@ on:
     types: [released]
 
 jobs:
-  build:
-      name: Publish release to PyPI
-      env:
-        PYPI_USERNAME_STSCI_MAINTAINER: ${{ secrets.PYPI_USERNAME_STSCI_MAINTAINER }}
-        PYPI_PASSWORD_STSCI_MAINTAINER: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }}
-        PYPI_USERNAME_OVERRIDE: ${{ secrets.PYPI_USERNAME_OVERRIDE }}
-        PYPI_PASSWORD_OVERRIDE: ${{ secrets.PYPI_PASSWORD_OVERRIDE }}
-        PYPI_TEST: ${{ secrets.PYPI_TEST }}
-        INDEX_URL_OVERRIDE: ${{ secrets.INDEX_URL_OVERRIDE }}
-      runs-on: ubuntu-latest
-      steps:
+  validate:
+    name: Validate metadata
+    runs-on: ubuntu-latest
+    steps:
+      - uses: spacetelescope/action-publish_to_pypi/validate@master
 
-          # Check out the commit containing this workflow file.
-          - name: checkout repo
-            uses: actions/checkout@v2
-         
-          - name: custom action
-            uses: spacetelescope/action-publish_to_pypi@master
-            id: custom_action_0
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    needs: [validate]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macos-10.15]
+
+    steps:
+      - uses: spacetelescope/action-publish_to_pypi/build-wheel@master
+
+  build_sdist:
+    name: Build source distribution
+    needs: [validate]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: spacetelescope/action-publish_to_pypi/build-sdist@master
+
+  upload_pypi:
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: spacetelescope/action-publish_to_pypi/publish@master
+        with:
+          test: ${{ secrets.PYPI_TEST }}
+          user: ${{ secrets.PYPI_USERNAME_STSCI_MAINTAINER }}
+          password: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }} # WARNING: Do not hardcode secret values here! If you want to use a different user or password, you can override this secret by creating one with the same name in your Github repository settings.
+          test_password: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER_TEST }}
```

Comment:
Does `test.pypi` actually get used for anything in these actions?

If so, we should also make sure that all these repos have corresponding spot on `test.pypi` and also that stsci_maintainer is a maintainer.  I don't think that's the case for `stsci.imagestats`.

Reply:
It does if `secrets.PYPI_TEST` is set, yes. And I agree, we should really grab the names while we can (why aren't packages namespaced by owner on PyPI??)

Another matter is that there is currently no `stsci_maintainer` account on `test.pypi.org` as far as I'm aware, and there's no `PYPI_PASSWORD_STSCI_MAINTAINER_TEST` secret in the `spacetelescope` organization either.
For my purposes I've just been overriding the secrets in my forked repository to use my own account, but I'll need to have those created, probably by @jneu since I don't have permission to do so at the moment.

Reply:
Sounds great.  And thanks for the extra info.  Automating delivery first to test.pypi and the real thing will be a huge time-saver for repos that do this manually, and I assume if the delivery to test.pypi fails, the workflow aborts then?  If so, that's a really nice feature.  We should document this somewhere for repo owners if possible.

But yeah, getting our org organized on both pypi and test.pypi are necessary.

Reply:
No, at the moment it publishes to pypi.org by default, unless the `PYPI_TEST` secret is set, in which case it publishes to test.pypi.org. This is mostly for testing the actual custom step and workflow itself; hopefully if it works for one version of one package, it should work for every version of every package, so we shouldn't need to publish every package to test.pypi.org on every run of the workflow.

That's not a bad idea though, if we run into publishing issues that only affect some packages and not others that would be a good piece of assurance to add.

Reply:
I see.  We've found that anytime we make major changes to package structure, how it is built, it's a good idea to deliver to test.pypi first.  I.e. when changing package metadata, etc.  Usually the `twine check --strict` will catch most things, but not always.

Reply:
Good to know, luckily we can add that functionality later within our custom step and roll it out when it's ready. Thanks for all the feedback!